src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -0,0 +1,131 @@
+import debounce from 'lodash.debounce';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Person } from '../../types/Person';
+import cn from 'classnames';
+
+interface Props {
+  people: Person[];
+  onSelectedPerson: (person: Person | null) => void;
+  delay?: number;
+}
+
+export const DropdownMenu: React.FC<Props> = ({
+  people,
+  onSelectedPerson,
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [filterQuery, setFilterQuery] = useState('');
+  const [activeDropdown, setActiveDropdown] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+  const applyQuery = useCallback(
+    debounce(string => {
+      setFilterQuery(string);
+    }, delay),
+    [],
+  );
+
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Consider following the naming conventions for event handler functions in React. For example, prefix the function name with 'handle' or 'on'.